asv_bench/benchmarks/period.py

```diff
@@ -3,141 +3,128 @@
+    def time_start_time(self, freq):
+        self.per.start_time
 
-    def time_drop_duplicates_pseries(self):
-        self.s.drop_duplicates()
+    def time_end_time(self, freq):
+        self.per.end_time
 
-    def time_drop_duplicates_pindex(self):
-        self.i.drop_duplicates()
 
-    def time_value_counts_pseries(self):
-        self.s.value_counts()
+class PeriodUnaryMethods(object):
+    params = ['M', 'min']
+    param_names = ['freq']
 
-    def time_value_counts_pindex(self):
-        self.i.value_counts()
+    def setup(self, freq):
+        self.per = Period('2012-06-01', freq=freq)
 
+    def time_to_timestamp(self, freq):
+        self.per.to_timestamp()
 
-class Properties(object):
-    def setup(self):
-        self.per = Period('2017-09-06 08:28', freq='min')
+    def time_now(self, freq):
+        self.per.now()
 
-    def time_year(self):
-        self.per.year
+    def time_asfreq(self, freq):
+        self.per.asfreq('A')
 
-    def time_month(self):
-        self.per.month
 
-    def time_day(self):
-        self.per.day
+class PeriodIndexConstructor(object):
+    goal_time = 0.2
 
-    def time_hour(self):
-        self.per.hour
+    params = ['D']
+    param_names = ['freq']
 
-    def time_minute(self):
-        self.per.minute
+    def setup(self, freq):
+        self.rng = date_range('1985', periods=1000)
+        self.rng2 = date_range('1985', periods=1000).to_pydatetime()
 
-    def time_second(self):
-        self.per.second
+    def time_from_date_range(self, freq):
+        PeriodIndex(self.rng, freq=freq)
 
-    def time_is_leap_year(self):
-        self.per.is_leap_year
+    def time_from_pydatetime(self, freq):
+        PeriodIndex(self.rng2, freq=freq)
 
-    def time_quarter(self):
-        self.per.quarter
 
-    def time_qyear(self):
-        self.per.qyear
+class DataFramePeriodColumn(object):
+    goal_time = 0.2
 
-    def time_week(self):
-        self.per.week
+    def setup_cache(self):
+        rng = pd.period_range(start='1/1/1990', freq='S', periods=20000)
+        df = pd.DataFrame(index=range(len(rng)))
+        return rng, df
 
-    def time_daysinmonth(self):
-        self.per.daysinmonth
+    def time_setitem_period_column(self, tup):
+        rng, df = tup
+        df['col'] = rng
 
-    def time_dayofweek(self):
-        self.per.dayofweek
 
-    def time_dayofyear(self):
-        self.per.dayofyear
+class Algorithms(object):
+    goal_time = 0.2
 
-    def time_start_time(self):
-        self.per.start_time
+    params = ['index', 'series']
+    param_names = ['typ']
 
-    def time_end_time(self):
-        self.per.end_time
+    def setup(self, typ):
+        data = [Period('2011-01', freq='M'), Period('2011-02', freq='M'),
+                Period('2011-03', freq='M'), Period('2011-04', freq='M')]
 
-    def time_to_timestamp():
-        self.per.to_timestamp()
+        if typ == 'index':
+            self.vector = PeriodIndex(data, freq='M')
```

Comment:
I would do here `data * 1000` as well like you did for series (to have more data)

Reply:
Happy to make that change, but the *1000 for series was there when I got here. Im implicitly assuming someone had a reason for doing it that way.

Reply:
I don't think there was a reason